examples/wasm/iwasmt.c

```diff
@@ -0,0 +1,165 @@
+{
+    /* exception memory is part of instance -> no buffer need */
+    const char *exception = 0;
+    int ret = 0;
+    wasm_application_execute_main(module_inst, argc, argv);
+    if ((exception = wasm_runtime_get_exception(module_inst))) {
+        puts(exception);
+        return -1;
+    }
+    if (argc > 0) {
+        ret = *((int*)argv);
+    }
+    return ret;
+}
+
+/* execs the main function of a loaded module */
+int iwasm_module_exec_main(wasm_module_t wasm_module, int argc, char **argv)
+{
+    wasm_module_inst_t wasm_module_inst = NULL;
+    int ret = 0;
+
+    {   /* instantiate the module */
+        char error_buf[128];
+        if (!(wasm_module_inst = wasm_runtime_instantiate(wasm_module, 8 * 1024,
+            8 * 1024, error_buf, sizeof(error_buf)))) {
+            puts(error_buf);
+            return -1;
+            }
+    }
+    ret = iwasm_instance_exec_main(wasm_module_inst, argc, argv);
+    /* destroy the module instance */
+    wasm_runtime_deinstantiate(wasm_module_inst);
+    return ret;
+}
+
+/* bytecode needs to be writeable*/
+int iwasm_bytecode_exec_main(uint8_t * bytecode, size_t bytecode_len, int argc, char **argv)
+{
+    int ret = 0;
+    wasm_module_t wasm_module = NULL;
+
+    {   /* load WASM module */
+        char error_buf[128];
+        if (!(wasm_module = wasm_runtime_load(bytecode, bytecode_len,
+                                          error_buf, sizeof(error_buf)))) {
+            puts(error_buf);
+            return -1;
+        }
+    }
+    ret = iwasm_module_exec_main(wasm_module, argc, argv);
+    /* unload WASM module */
+    wasm_runtime_unload(wasm_module);
+    return ret;
+}
+
+/* running this initialises the wamr runtime*/
+bool iwasm_runtime_init(void)
+{
+    RuntimeInitArgs init_args;
+/* chose allocator non defaults to system allocator */
+#define FUNC_ALLOC
+// #define POOL_ALLOC
+
+    memset(&init_args, 0, sizeof(RuntimeInitArgs));
+#ifdef POOL_ALLOC
+    static char global_heap_buf[256 * 1024] = { 0 };//(256 kB)
+
+    init_args.mem_alloc_type = Alloc_With_Pool;
+    init_args.mem_alloc_option.pool.heap_buf = global_heap_buf;
+    init_args.mem_alloc_option.pool.heap_size = sizeof(global_heap_buf);
+#elif defined(FUNC_ALLOC)
+    init_args.mem_alloc_type = Alloc_With_Allocator;
+/*deactivate Wpedantic for some lines*/
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+    init_args.mem_alloc_option.allocator.malloc_func = malloc;
+    init_args.mem_alloc_option.allocator.realloc_func = realloc;
+    init_args.mem_alloc_option.allocator.free_func = free;
+#pragma GCC diagnostic pop
+#else
+    init_args.mem_alloc_type = Alloc_With_System_Allocator;
+#endif
+
+    /* initialize runtime environment */
+    if (!wasm_runtime_full_init(&init_args)) {
+        puts("Init runtime environment failed.");
+        return false;
+    }
+    return true;
+}
+
+void iwasm_runtime_destroy(void)
+{
+    wasm_runtime_destroy();
+
+}
```

Comment:
```suggestion
{
    wasm_runtime_destroy();
}
```

Reply:
Why is this indirection needed?

Reply:
i want to avoid the  #include "wasm_export.h" in wasm-main.c  -> one can take iwasm.c and the inline headers in wasm-main.c and just exec some other bytecode by adding some other blobs.
ease the example remixing